nanoFramework.Logging.Syslog/SyslogClient.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace nanoFramework.Logging.Syslog
+{
+    /// <summary>
+    /// Syslog Client that sends logs to a syslog server following RFC3164 format.
+    /// </summary>
+    public class SyslogClient : IDisposable
+    {
+        private readonly Socket _socket;
+        private bool _disposed;
+
+        /// <summary>
+        /// Default Syslog UDP port : 514
+        /// </summary>
+        public const int DefaultPort = 514;
+
+        /// <summary>
+        /// Create a SyslogClient instance that will send RFC3164 compliant messages to a syslof server
+        /// </summary>
+        /// <param name="endpoint">Endpoint of the server</param>
+        /// <param name="localHostname">'Hostname' part of the RFC3164 message</param>
+        /// <param name="facility">Facility used by this logger</param>
+        /// <param name="localAddress">Local IP address to bind socket (if null IPAdress.Any will be used)</param>
+        /// <param name="localPort">Local port to bind socket (0 to choose available port)</param>
+        public SyslogClient(IPEndPoint endpoint, string localHostname, Facility facility = default, IPAddress localAddress = null, int localPort = 0)
+        {
+            if (localHostname is null)
+                throw new ArgumentNullException(nameof(localHostname));
```

Comment:
We are using brakets in all the code for the if statements. now, there is a better way to handle this one, you can replace this if statement and the LoclHostname line by:
```csharp
LocalHostname = localHostname ?? throw new ArgumentNullException(nameof(localHostname));
```